shells/lib/components/context/context-stores.js

```diff
@@ -46,6 +46,7 @@ const ContextStoresImpl = class {
     // cache and return promises in case of re-entrancy
     let promise = pendingStores[id];
     if (!promise) {
+      // eslint-disable-next-line no-async-promise-executor
```

Comment:
I've read https://eslint.org/docs/rules/no-async-promise-executor and it said that pretty much all async executor functions can and should be avoided. I went through a number of them in the previous pass and it appeared correct - I suggested how we can change them to not be async. I suggested that we should fix the code in remaining places, can we do this? There are very few places remaining,